routes/user.playlist.js

```diff
@@ -252,139 +255,210 @@ router.route('/:playlist_name/export3')
-                        }
+                        return playlist.translationsChecked
+                            // minLine: _.min(playlist.translationsChecked.map((t) => t.lyric.length))
+
                     })
                     // songs2d = playlists.map((playlist) => playlist.translationsChecked)
                 next()
             })
     })
     .get(function(req, res, next) {
-        var type = req.query.type;
+        if (type == 1) {
+            console.log('pptx1')
+            var newSongsArr = create2dArrayOfOneSong(songs2d)
+            res.render('export/preview-ppt1', {
+                songs2d: newSongsArr
+            })
+        } else if (type == 2) {
+            var newSongsArr = create2dArrayOfTwoSongs(songs2d)
+            res.render('export/preview-ppt2', {
+                songs2d: newSongsArr
+            })
+        } else if (type == 3) {
+            res.render('export/preview-ppt3', {
+                songs2d: songs2d
+            })
+        } else {
+            console.log('pdf')
+            res.render('export/preview-pdf', {
+                songs2d: songs2d
+            })
+        }
+    })
+    .post(function(req, res, next) {
         var filename;
-        if (type === 'pdf') {
+        // if (type === '') {
+        //
+        // } else
+        if (type == 1) {
+            // var newSongsArr = create2dArrayOfTwoSongs(songs2d)
+            // generateSlideByStanza(res, newSongsArr, playlistName, 2)
+        } else if (type == 2) {
+            // var newSongsArr = create2dArrayOfTwoSongs(songs2d)
+            // generateSlideByStanza(res, newSongsArr, playlistName, 2)
+        } else if (type === 'pptx' && langPerSlide === 3) {
+            // var newSongsArr = create2dArrayOfTwoSongs(songs2d)
+            // generateSlideByStanza(res, newSongsArr, playlistName, 2)
+        } else {
             filename = Date.now()
-            app.render('handout', {
+            app.render('export/handout', {
                 songs2d: songs2d
             }, function(err, html) {
-                console.log(html)
                 pdf.create(html).toStream(function(err, stream) {
                     res.setHeader('Content-Type', 'application/pdf')
                     res.setHeader('Content-Disposition', 'attachment; filename=' + filename + '.pdf')
                     stream.pipe(res)
                 })
             });
-        } else if (type === 'pptx') {
-            filename = Date.now()
-            var pptx = officegen('pptx');
+        }
+    })
 
-            var slide;
-            var pObj;
+module.exports = router;
 
-            pptx.on('finalize', function(written) {
-                console.log('Finish to create a PowerPoint file.\nTotal bytes created: ' + written + '\n');
-            });
 
-            pptx.on('error', function(err) {
-                console.log(err);
+//this function will generate multiple pptx files, compressed it to zip, and pipe it to client
+/*
```

Comment:
If you want to use multiline comments, each line should have one *.
The first and last line of the comment might have /** **/.